src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java

```diff
@@ -172,8 +173,8 @@ private void runGcloudCommand(List<String> args, String... topLevelCommand)
   // class's main configured ProcessRunner should be used.
   private String runSynchronousGcloudCommand(List<String> args)
       throws ProcessRunnerException {
-    validateCloudSdk();
-
+    validateCloudSdkLocation();
```

Comment:
I'm not a big fan of the way we have the hidden runtime validation exceptions that can blow up callers of this. Not really this PRs problem though.